.github/workflows/rust.yml

```diff
@@ -60,3 +60,47 @@ jobs:
         run: cargo test
       - name: Run clippy
         run: cargo clippy # -- -D warnings
+
+  coverage:
+    name: Coverage
+    needs: find-crates
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        directory: ${{ fromJSON(needs.find-crates.outputs.crates) }}
+        exclude:
+          - directory: hoverkite-firmware
+    defaults:
+      run:
+        working-directory: ${{ matrix.directory }}
+    env:
+      RUSTC_BOOTSTRAP: 1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: sudo apt-get install libudev-dev
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: thumbv7m-none-eabi
+      - name: Install grcov
+        run: curl -L https://github.com/mozilla/grcov/releases/latest/download/grcov-linux-x86_64.tar.bz2 | tar jxf -
+      - name: Install llvm-tools
+        run: rustup component add llvm-tools-preview
+      - name: Build for coverage
+        run: cargo build --all-features
+        env:
+          RUSTFLAGS: "-Zinstrument-coverage"
```

Comment:
(optional) slight preference for env before run, because it's the order that you would see it in bash script

Reply:
I did it in this order in all our other projects, so I think I'd rather keep it the same for consistency. (Or change it everywhere, but I'm lazy.)